### ISSUE TITLE
FeatureInfo: upgrade to ReactMarkdown 6.0

### ISSUE THREAD

**Issue by jacobwod** [hajkmap/Hajk#761]
_Edit: originally this issue was only about improving the typography in FeatureInfo, but it turned out to be something bigger. I change the title so it reflects the ongoing effort of upgrading to a newer version of our Markdown library._

### Current status
[This branch](features/761-upgrade-react-markdown-to-v6) houses the almost complete upgrade. There's still no way to solve transforming our custom syntax (`{foo@@bar}`). I'm putting this on hold for now, hoping for a solution in [this discussion](https://github.com/remarkjs/remark/discussions/700). 

## Original issue
There are a couple of improvements to be made to the Markdown parser's mapping to MUI Typograhpy.

### Add some spacing after Heading
![Unknown](https://user-images.githubusercontent.com/110222/115348218-a4703600-a1b2-11eb-8eae-db6be866c07f.png)

### Remove spacing prior to Paragraph
![Unknown](https://user-images.githubusercontent.com/110222/115348227-a76b2680-a1b2-11eb-8034-5482d74b7eab.jpeg)

**Comment by jacobwod** [comment]
I think I'll do those changes when I lift react-markdown to 6.0.0, there are some API changes, see https://github.com/remarkjs/react-markdown/blob/main/changelog.md#600---2021-04-15.

**Comment by jacobwod** [comment]
Turns out that this breaks the rendering of externally fetched components (like `{foobar@@documenthandler}`). 

The reason is that react-markdown as of 6.0.0 doesn't have the "hook" that I relied on previously for converting remaining placeholders to externally fetched components. 

[I've asked](https://github.com/remarkjs/remark/discussions/700) how to go on from here, we'll see where the discussion leads.

**Reopening and labelling as bug.**
